shopping_mall/backend/tests/conftest.py

```
@@ -0,0 +1,245 @@
+    """시나리오 기반 AgentClient 스텁.
+
+    responses 리스트를 순서대로 반환하며, AgentUnavailableError를 포함시키면
+    해당 호출에서 예외를 발생시킵니다.
+    """
+
+    def __init__(self, responses: list):
+        self._responses = iter(responses)
+        self.calls: list[list[dict]] = []  # chat_with_tools 호출당 messages 스냅샷
+
+    async def chat_with_tools(
+        self, messages: list[dict], tools: list[dict], system: str
```

Ruff (0.15.10)

[warning] Unused method argument: `tools`

(ARG002)